spyder/plugins/completion/kite/widgets/status.py

```diff
@@ -16,50 +16,52 @@
 from spyder.utils import icon_manager as ima
 from spyder.widgets.calltip import ToolTipWidget
 from spyder.widgets.status import StatusBarWidget
-from spyder.plugins.completion.kite.utils.status import NOT_INSTALLED
+from spyder.plugins.completion.kite.utils.status import (
+    check_if_kite_installed, NOT_INSTALLED)
 
 logger = logging.getLogger(__name__)
 
 
 class KiteStatusWidget(StatusBarWidget):
     """Status bar widget for Kite completions status."""
     BASE_TOOLTIP = _("Kite completions status")
-    DEFAULT_STATUS = _('not reacheable')
+    DEFAULT_STATUS = _('not reachable')
 
     def __init__(self, parent, statusbar, plugin):
         self.plugin = plugin
         self.tooltip = self.BASE_TOOLTIP
         super(KiteStatusWidget, self).__init__(parent, statusbar,
                                                icon=ima.get_kite_icon())
-        self.tooltipWidget = ToolTipWidget(self, as_tooltip=True)
-        self.set_value(None)
+        self.tooltip_widget = ToolTipWidget(self, as_tooltip=True)
+        is_installed, _ = check_if_kite_installed()
+        self.setVisible(is_installed)
 
     def set_value(self, value):
         """Return Kite completions state."""
-        kite_enabled = self.plugin.get_option('enable')
+        is_installing = self.plugin.is_installing()
+        installation_status = self.plugin.installation_cancelled_or_errored()
         if (value is not None and 'short' in value):
             self.tooltip = value['long']
             value = value['short']
-        elif value is not None and self.plugin.is_installing():
-            self.tooltip = _('Kite is being installed')
+        elif value is not None and is_installing or installation_status:
+            self.setVisible(True)
             if value == NOT_INSTALLED:
                 return
         elif value is None:
             value = self.DEFAULT_STATUS
             self.tooltip = self.BASE_TOOLTIP
-
         self.update_tooltip()
-        self.setVisible(value != NOT_INSTALLED or kite_enabled)
-
+        self.setVisible(value != NOT_INSTALLED)
```

Comment:
I agree with hiding the widget if kite is not enabled (as it was before this change).